packages/data/addon/services/cardstack-data.js

```diff
@@ -0,0 +1,96 @@
+import Service from '@ember/service';
+import { inject } from '@ember/service';
+import { camelize } from '@ember/string';
+import { singularize, pluralize } from 'ember-inflector';
+import DS from 'ember-data';
+
+const noFields = '___no-fields___';
+
+export default Service.extend({
+  store: inject(),
+
+  init() {
+    this._super();
+    this._loadedRecordsCache = {};
+    this._contentTypeCache = {};
+  },
+
+  async load(type, id, format) {
```

Comment:
This API will eventually also want `branch` as an argument.

Reply:
I'll add that parameter in there now so it wont be a mess to deal with in the future

Reply:
or well... when we perform `store.findRecord()` how do we incorporate `branch` into that request? or is that not currently supported?

Reply:
Nevermind, found it: https://github.com/cardstack/cardstack/blob/master/packages/routing/addon/routes/cardstack/common.js#L32